dec/decode.go

```diff
@@ -169,10 +178,11 @@ func NewBrotliReader(stream io.Reader) *BrotliReader {
 func NewBrotliReaderSize(stream io.Reader, size int) *BrotliReader {
 	r := &BrotliReader{
 		reader: stream,
+		state:  new(C.BrotliState),
```

Comment:
This is presumably still going to cause GC panics (#14) when the C struct is traversed?
Should we replace this with []byte and unsafe pointers?